src/TraceEvent/BPerf/BPerfEventSource.cs

```diff
@@ -0,0 +1,312 @@
+﻿//     Copyright (c) Microsoft Corporation.  All rights reserved.
+// This file is best viewed using outline mode (Ctrl-M Ctrl-O)
+//
+// This program uses code hyperlinks available as part of the HyperAddin Visual Studio plug-in.
+// It is available from http://www.codeplex.com/hyperAddin 
+// 
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using Microsoft.Diagnostics.Tracing.Parsers;
+using Microsoft.Diagnostics.Tracing.Parsers.Kernel;
+
+namespace Microsoft.Diagnostics.Tracing
+{
+    public sealed class BPerfEventSource : TraceEventDispatcher, IDisposable
```

Comment:
Can you add some comments to describe what's happening in this file?  Would love to have a class-level comment at a minimum that describes the overall algorithm here.

Reply:
Yup will do.